helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -28,24 +38,116 @@
  * A placeholder before we have the real assignment metadata store.
  */
 public class AssignmentMetadataStore {
-  private Map<String, ResourceAssignment> _persistGlobalBaseline = new HashMap<>();
-  private Map<String, ResourceAssignment> _persistBestPossibleAssignment = new HashMap<>();
+  private static final String ASSIGNMENT_METADATA_KEY = "ASSIGNMENT_METADATA";
+  private static final String BASELINE_TEMPLATE = "/%s/%s/BASELINE";
+  private static final String BEST_POSSIBLE_TEMPLATE = "/%s/%s/BEST_POSSIBLE";
+  private static final String BASELINE_KEY = "BASELINE";
+  private static final String BEST_POSSIBLE_KEY = "BEST_POSSIBLE";
+  private static final ZkSerializer SERIALIZER = new ZNRecordJacksonSerializer();
+
+  private BucketDataAccessor _dataAccessor;
+  private String _baselinePath;
+  private String _bestPossiblePath;
+  private Map<String, ResourceAssignment> _globalBaseline;
+  private Map<String, ResourceAssignment> _bestPossibleAssignment;
+
+  public AssignmentMetadataStore(HelixManager helixManager) {
+    _dataAccessor = new ZkBucketDataAccessor(helixManager.getMetadataStoreConnectionString());
+    _baselinePath =
+        String.format(BASELINE_TEMPLATE, helixManager.getClusterName(), ASSIGNMENT_METADATA_KEY);
```

Comment:
Will it be cleaner if we make 3 methods instead? And then just call these methods whenever required.
1. getAssignmentMetadatPath(clusterName)
2. getBaselinePath(metadataPath)
3. getBestPossiblePath(metadataPath)

Reply:
@jiajunwang I want to make the paths static for now instead of generating it every time. It's not too clear whether we'll ever have growing number paths we write to but for now there are only two. Hard-coding gets the job done without incurring any more work.